analyses/cnv-chrom-plot/util/bin-coverage.R

```diff
@@ -0,0 +1,143 @@
+# Functions for calling CN statuses of genome bins
+#
+# C. Savonen for ALSF - CCDL
+#
+# 2020
+
+bp_per_bin <- function(bin_ranges, status_ranges) {
+  # Given a binned genome ranges object and another GenomicRanges object, 
+  # Return the number of bp covered per bin.
+  #
+  # Args:
+  #   bin_ranges: A binned GenomicRanges made from tileGenome. 
+  #   status_ranges: A GenomicRanges object to calculate what percent coverage of
+  #   each bin. 
+  #
+  # Returns:
+  #  a data.frame with bins x number of bp 
+  
+  # Find the portions of each segment that overlap with each bin.
+  bin_overlaps <- GenomicRanges::pintersect(
+    IRanges::findOverlapPairs(
+      bin_ranges,
+      status_ranges
+    )
+  )
+  
+  # Which segs are a part of which bins?
+  bin_indices <- GenomicRanges::findOverlaps(
+    bin_ranges,
+    bin_overlaps
+  )
+  
+  # Get the sum of the length of all seg portions for each bin.
+  bp_per_bin <- tapply(
+    bin_overlaps@ranges@width, # Get length of each sequence within the bin
+    bin_indices@from, # Index of which bin it overlaps
+    sum
+  ) # Add up length per bin
+  
+  # Format as data.frame with rows = bins
+  per_bin_df <- data.frame(
+    bin = as.numeric(names(bp_per_bin)),
+    bp_per_bin = as.numeric(bp_per_bin)
+  )
+  
+  # Store dummy counts if there are no ranges that are in the bins
+  if (nrow(per_bin_df) == 0) {
+    per_bin_df <- data.frame(
+      bin = as.numeric(1:length(bin_ranges)),
+      bp_per_bin = 0
+    )
+  }
+  return(per_bin_df)
+}
+
+call_bin_status <- function(sample_id,
+                            seg_ranges,
+                            bin_ranges,
+                            perc_delta_threshold) {
+  # Given a sample_id, CN segment ranges, and binned genome ranges object, 
+  # make a call for each bin on what CN copy status has the most coverage in the bin. 
+  # Uses bp_per_bin function. 
+  #
+  # Args:
+  #   sample_id: A string that corresponds to the biospecimen s
```

Comment:
The function description above makes me believe that the `sample_id` argument should be provided a string with a single biospecimen id, is this the case?

Reply:
Yes. I will make this clarification!